enhancements/ingress/gateway-api-with-cluster-ingress-operator.md

```diff
@@ -790,7 +790,32 @@ To answer this question, we need to understand the requirements around feature
 gates and understand whether just installing the gatewayclasses CRD may be
 problematic for some clusters.
 
-**Resolution**: TBD.
+**Resolution**: Yes, we should use a feature gate.  It is not just the installation of
+the CRDs that must be considered.  The additional controllers and watchers needed for
+managing Gateway API resources should be avoided if not needed. In addition, having an
+official feature gate is useful for setting expectations about whether a cluster may be
+upgraded. In Dev and Tech Preview, a cluster with this feature should not be upgradeable.
+Finally, if a user wants to install the Gateway API CRDs on their cluster for experimenting
+with an implementation other than OSSM/Istio, we should permit them to do this without
+interference from OpenShift Gateway API on OSSM.
+
+In Dev Preview we choose to use the `CustomNoUpgrade` variety of feature gate primarily
+because our other choice was named `TechPreviewNoUpgrade`, and we felt this would cause some
+confusion over the status of the feature.  Even though `TechPreviewNoUpgrade` has automatic
+CI testing in place and is managed by the OpenShift API, we won't need it until the feature has
+graduated in maturity to Tech Preview.
+
+The list of feature gate constraints includes:
+
+* The feature gate should be named `GatewayAPI` for all iterations of the feature.
+* The GatewayAPI feature gate should be a `CustomNoUpgrade` variety in Dev Preview,
+a `TechPreviewNoUpgrade` variety in Tech Preview, and removed before GA.
+* The GatewayAPI feature gate will be used to bar the installation of the GatewayAPI CRDs.
+* The GatewayAPI feature gate will be used to bar the installation, configuration, or watch of OpenShift Service Mesh
+components.
+* The GatewayAPI feature gate will be used to bar special operation of the LoadBalancer and DNS.
+* The GatewayAPI feature gate will be used to lift the restriction of DNSRecords to an owning
+ingresscontroller.
```

Comment:
Is this an important restriction?  It would complicate the implementation—not much, but without clear benefit.

Reply:
I thought this was an important feature of the GWAPI work, that we had to enable DNS records being created by something other than an owning ingresscontroller.

Reply:
The DNS management is important, but restricting management to DNSRecord CRs that have an owning IngressController seems unnecessary.  The current implementation simply deletes that restriction: https://github.com/openshift/cluster-ingress-operator/commit/18337156bdd5a2c0ff9d4e7f219b8e4fd0105540

That is, we had this logic:

1. Get the DNSRecord CR.
2. If no owning IngressController is found, then don't reconcile.
3. Otherwise, reconcile the DNSRecord.

The paragraph that you wrote seems to be saying that the logic is changed to the following:

1. Get the DNSRecord CR.
2. If the "GatewayAPI" feature isn't enabled and no owning IngressController is found, then don't reconcile.
3. Otherwise, reconcile the DNSRecord.

Instead, I changed to logic so it always reconciles the DNSRecord CR irrespective of any owner.  

If it is important to keep the restriction when the "GatewayAPI" feature isn't enabled, then I can add restore the original logic and add a check for the feature, but it seems to me that that adds unnecessary complexity.

Reply:
I'll clarify.  I definitely did not mean what you thought.

Reply:
I didn't know that this applies whether there is a feature gate or not, so I removed the line.